### PR TITLE
feat: add support for inter-cluster communication for Ray plugin

### PR DESCRIPTION
**Pull request by aybidi**
# TL;DR
This PR addresses the concern "[Supporting customized k8s client in various k8s plugins](https://github.com/flyteorg/flyte/issues/2883)" but narrows it down to the ray plugin. In short, users can configure ray plugin to do inter-cluster communication, essentially submitting a custom resource to a specified k8s cluster instead of the one flytepropeller runs on.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue

https://github.com/flyteorg/flyte/issues/2883

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: go/tasks/plugins/k8s/ray/config.go [hunk 1]
```diff
@@ -40,6 +94,9 @@ type Config struct {
 
 	// NodeIPAddress the IP address of the head node. By default, this is pod ip address.
 	NodeIPAddress string `json:"nodeIPAddress,omitempty"`
+
+	// Remote Ray Cluster Config
+	RemoteClusterConfig ClusterConfig `json:"remoteClusterConfig" pflag:"Configuration of remote K8s cluster for array jobs"`
```
Comment (pingsutw) [inline go/tasks/plugins/k8s/ray/config.go:99]:
```suggestion
	RemoteClusterConfig ClusterConfig `json:"remoteClusterConfig" pflag:"Configuration of remote K8s cluster for ray jobs"`
```

File: go/tasks/plugins/k8s/ray/config.go [hunk 2]
```diff
@@ -1,11 +1,65 @@
...
+	"io/ioutil"
+
 	pluginsConfig "github.com/flyteorg/flyteplugins/go/tasks/config"
+	"github.com/pkg/errors"
+	restclient "k8s.io/client-go/rest"
 )
 
 //go:generate pflags Config --default-var=defaultConfig
 
+type ClusterConfig struct {
```
Comment (hamersaw) [inline go/tasks/plugins/k8s/ray/config.go:14]: Does it makes sense to move the `ClusterConfig` and `Auth` structs, probably the `KubeClientConfig` func as well, over to the `go/tasks/pluginmachinery/k8s` directory?

Then adding this support to all k8s plugins would be as simple as adding the `ClusterConfig` to the plugin configuration and setting the `CustomKubeClient` on the `PluginEntry` during registration.

Comment (aybidi) [reply go/tasks/plugins/k8s/ray/config.go:14]: That would be the best-case scenario. We tried narrowing the focus to ray plugin to make things fast. Ideally, specifying a custom kubeclient will be possible for all k8s plugins. I'll refactor `ClusterConfig` and `Auth` structs to be under `go/tasks/pluginmachinery/k8s`.

Comment (aybidi) [reply go/tasks/plugins/k8s/ray/config.go]: PTAL

**Review by hamersaw — APPROVED** [review]
